roles/openshift_health_checker/openshift_checks/efk_stack.py

```diff
@@ -0,0 +1,604 @@
+
+        # TODO(lmeyer): run it all again for the ops cluster
+        return {"failed": False, "changed": False, "msg": 'No problems found with logging stack.'}
+
+    def _get_pods(self, task_vars):
+        '''Get all pods and sort them by EFK component. Returns: pods by component, error string'''
+        pod_output = self._exec_oc("get pods -o json", [], task_vars)
+        try:
+            pods = json.loads(pod_output)
+            if not pods or not pods.get('items'):
+                raise ValueError()
+        except ValueError:
+            # successful run but non-parsing data generally means there were no pods in the namespace
+            return None, 'There are no pods in the {} namespace. Is logging deployed?'.format(self.logging_namespace)
+
+        logging_components = ['es', 'fluentd', 'kibana', 'curator']
+        pods_for = {comp: [] for comp in logging_components}
+
+        for pod in pods['items']:
+            component = pod['metadata']['labels'].get('component')
+            if component in logging_components:
+                pods_for[component].append(pod)
+
+        return pods_for, None
+
+    @staticmethod
+    def _not_running_pods(pods):
+        '''Returns: list of pods not in a ready and running state'''
+        return [
+            pod for pod in pods
+            if any(
+                container['ready'] is False
+                for container in pod['status']['containerStatuses']
+            ) or not any(
+                condition['type'] == 'Ready' and condition['status'] == 'True'
+                for condition in pod['status']['conditions']
+            )
+        ]
+
+    def _running_es_pods(self, es_pods):
+        '''Returns: list of running pods, list of errors about non-running pods'''
+        not_running = self._not_running_pods(es_pods)
+        if not_running:
+            return [pod for pod in es_pods if pod not in not_running], [(
+                'The following Elasticsearch pods are not running:\n'
+                '{pods}'
+                'These pods will not aggregate logs from their nodes.'
+            ).format(pods=''.join(
+                "  {} ({})\n".format(pod['metadata']['name'], pod['spec'].get('host', 'None'))
+                for pod in not_running
+            ))]
+        return es_pods, []
+
+    def check_elasticsearch(self, es_pods, task_vars):
+        '''Various checks for elasticsearch. Returns: error string'''
+        running_pods, error_msgs = self._running_es_pods(es_pods)
+        pods_by_name = {
+            pod['metadata']['name']: pod for pod in running_pods
+            # Filter out pods that are not members of a DC
+            if pod['metadata'].get('labels', {}).get('deploymentconfig')
+        }
+        if not pods_by_name:
+            return 'No logging Elasticsearch pods were found. Is logging deployed?'
+        error_msgs += self._check_elasticsearch_masters(pods_by_name, task_vars)
```

Comment:
What's "this"? If there are no pods running the `return` above aborts any further testing.